pkg/blockstorage/azure/azuredisk.go

```diff
@@ -44,6 +44,22 @@ func (s *AdStorage) Type() blockstorage.Type {
 	return blockstorage.TypeAD
 }
 
+// Authenticate check Azure creds if the credType is supported
+func (s *AdStorage) Authenticate(ctx context.Context, credType string, creds map[string]string) (CredsValidity, error) {
```

Comment:
This interface is not united tested because creating a new Azure provider involves calling `Azure Instance Metadata` which will results in `context timeout`